### PR TITLE
backend-*-api: add RedactionsService

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

I realized that customizing the root logger is kinda tricky right now since it also wires up the secret redactions from config. That and the follow-up for https://github.com/backstage/backstage/pull/24478 made me think that it might be worth introducing a redactions service. It both lets plugins add sensitive data to redactions and also lets them do their own redactions for sensitive content. In particular, I'm thinking that both Scaffolder and TechDocs build logs that we stream to the client could use this.

Kept it very simple for now, but there's one problem that might be worth introducing some extra solution for, which is that an ever-growing redactions filter might have a performance impact. A long-running deployment might simply pile up a lot of secrets over time to the point where the RegExp compilation and execution becomes a problem. There's also the potential issue of ReDos attacks if user input is ever forwarded to the redactions by a plugin.

One potential solution is to be able to add a redaction with a TTL, although there's a risk that it only solves the long-running service problem and not the malicious user input. Another option could be the creation of some form of redaction context where secrets can be added only for that particular context, the idea being that this is particularly useful to handle scaffolder user input. That might be a bit overkill for a service though, and it's possible that we should instead implement local filtering in the scaffolder in addition to using the service.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by benjdlambert — APPROVED** [review]

**Comment by benjdlambert** [comment]
@Rugvip this looks like it needs a rebase 😅

File: .changeset/hip-starfishes-prove.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  deps: {
+    redactions: coreServices.redactions,
+  },
+  async factory({ redactions }) {
+    const source = ConfigSources.default({
+      argv: options?.argv,
+      remote: options?.remote,
+      watch: options?.watch,
+    });
+[
```
Comment (benjdlambert) [inline .changeset/hip-starfishes-prove.md:23]:
```suggestion

```

**Review by benjdlambert — APPROVED** [review]
Just a small thing on the changeset

**Comment by camilaibs** [comment]
Related: https://github.com/backstage/backstage/pull/26532

**Pull request closed without merging**